WickedEngine/wiLua.cpp

```diff
@@ -42,25 +42,64 @@ namespace wi::lua
 		}
 	};
 	LuaInternal luainternal;
-	std::string script_path;
+
+	uint32_t Internal_GenScriptPID(){
+		static std::atomic<uint32_t> scriptpid_next{ 0 + 1 };
+		return scriptpid_next.fetch_add(1);
+	}
+
+	uint32_t Internal_EncapsulateScript(std::string& script, std::string filename = "", uint32_t PID = 0){
+		static const std::string persistent_inject = R"(
+			local runProcess = function(func) 
+				success, co = Internal_runProcess(SCRIPT_FILE(), SCRIPT_PID(), func)
+				return success, co
+			end
+			if _ENV.PROCESSES_DATA[SCRIPT_PID()] == nil then
+				_ENV.PROCESSES_DATA[SCRIPT_PID()] = { _INITIALIZED = -1 }
+			end
+			if _ENV.PROCESSES_DATA[SCRIPT_PID()]._INITIALIZED < 1 then
+				_ENV.PROCESSES_DATA[SCRIPT_PID()]._INITIALIZED = _ENV.PROCESSES_DATA[SCRIPT_PID()]._INITIALIZED + 1
+			end
+		)";
+
+		if(PID == 0){
+			PID = Internal_GenScriptPID();
+		}
+
+		std::string dynamic_inject = "local function SCRIPT_FILE() return \""+filename+"\" end\n";
+		dynamic_inject += "local function SCRIPT_PID() return \""+std::to_string(PID)+"\" end\n";
```

Comment:
Why after `return there is a `\` character? I think this causes error on Windows, I don't think this is needed.

Reply:
Oh, it's for the `"`, I see